scripts/azureml-assets/azureml/assets/environment/pin_image_versions.py

```diff
@@ -36,6 +36,22 @@ def _urlopen_with_retries(request: Union[Request, str]) -> HTTPResponse:
     return urlopen(request)
 
 
+def _get_manifest(tag: str, hostname: str, repo: str):
+    """Retrieve manifest for an image.
+
+    Args:
+        tag (str): Tag for the image.
+        hostname (str): Hostname of the container registry.
+        repo (str): Repository of the image.
+    """
+    encoded_tag = urllib.parse.quote(tag, safe="")
+    request = Request(f"https://{hostname}/v2/{repo}/manifests/{encoded_tag}",
+                      method="HEAD",
+                      headers={'Accept': "application/vnd.docker.distribution.manifest.v2+json"})
+
+    return request

```

Comment:
How about just calling _urlopen_with_retries from here? I don't think there's a need to ever create a request without also executing it.